compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -681,14 +681,17 @@ class SpaceEngine(implicit ctx: Context) extends SpaceLogic {
       }
     }
 
+    val removeThisType = new RemoveThisMap
+    val tvars = tp1.typeParams.map { tparam => newTypeVar(tparam.paramInfo.bounds) }
+    val protoTp1 = removeThisType.apply(tp1).appliedTo(tvars)
+
     val force = new ForceDegree.Value(
-      tvar => !(ctx.typerState.constraint.entry(tvar.origin) eq tvar.origin.underlying),
+      tvar =>
+        !(ctx.typerState.constraint.entry(tvar.origin) eq tvar.origin.underlying) ||
+        (tvar `eq` removeThisType.prefixTVar),
```

Comment:
Why a backquoted `eq` here (or elsewhere for that matter)?

Reply:
I should use it in both cases, just following a coding style experimented by Martin in Dotty.